projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-users-endpoint.php

```diff
@@ -181,14 +181,16 @@ public function callback( $path = '', $blog_id = 0 ) {
 		) : array();
 		$viewers = array_map( array( $this, 'get_author' ), $viewers );
 
-		// we restrict search field to name when include_viewers is true.
+		// When include_viewers is true, search by username or email.
 		if ( $include_viewers && ! empty( $args['search'] ) ) {
 			$viewers = array_filter(
 				$viewers,
 				function ( $viewer ) use ( $args ) {
+					// Convert to WP_User so expected fields are available.
+					$wp_viewer = new WP_User( $viewer->ID );
 					// remove special database search characters from search term
 					$search_term = str_replace( '*', '', $args['search'] );
-					return strpos( $viewer->name, $search_term ) !== false;
+					return ( strpos( $wp_viewer->user_login, $search_term ) !== false || strpos( $wp_viewer->user_email, $search_term ) !== false );
```

Comment:
Do we need to take into account use cases where `$wp_viewer` would have no or empty `user_login` and `user_email`?

Reply:
I'm not sure! I suppose to be on the safe side we can also preserve the current check for `name` - the corresponding `WP_User` property is `display_name`.